lib/processor.js

```diff
@@ -207,17 +216,41 @@ function preprocess(text) {
 function postprocess(messages) {
 
     /**
+     * Determine the number of messages per line
+     * @param   {Message[]} msgs A flattened array of messages with mapped locations.
+     * @returns {integer[]}      A hash of line numbers and message quantities.
+     */
+    function getErrorsPerLine(msgs) {
+        return msgs.reduce(function(acc, error) {
+            acc[error.line] = acc[error.line] || 0;
+            acc[error.line] += 1;
+            return acc;
+        }, {});
+    }
+
+
+    /**
      * Returns messages which are not in the list of expected errors
      * @param {Message[]} msgs A flattened array of messages with mapped locations.
      * @returns {Message[]} An array of messages without expected messages.
      */
     function getUnexpectedMessages(msgs) {
-        return msgs.filter(function(error) {
+        var lineMsgQtys = getErrorsPerLine(msgs);
+        var filteredMsgs = msgs.filter(function(error) {
             if (!expectedErrors[error.line]) {
                 return true;
             }
             return (expectedErrors[error.line].indexOf(error.message) === -1);
         });
+
+        return filteredMsgs.filter(function(error) {
```

Comment:
I'd merge this and the previous `filter` to avoid iterating through twice. There should be a sensible, legible way to structure the checks.